### PR TITLE
Prepare rc16 release

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 [![codecov](https://codecov.io/gh/arturbosch/detekt/branch/master/graph/badge.svg)](https://codecov.io/gh/arturbosch/detekt)
 [![Download](https://api.bintray.com/packages/arturbosch/code-analysis/detekt/images/download.svg) ](https://bintray.com/arturbosch/code-analysis/detekt/_latestVersion)
 <!-- Version needs to be hardcoded until rc phase is over. Change to '-'RC for semantic versioning broke the badge. [![gradle plugin](https://img.shields.io/maven-metadata/v/https/plugins.gradle.org/m2/io/gitlab/arturbosch/detekt/detekt-gradle-plugin/maven-metadata.xml.svg?label=gradle&style=flat-square)](https://plugins.gradle.org/plugin/io.gitlab.arturbosch.detekt)-->
-[![gradle plugin](https://img.shields.io/badge/Gradle-1.0.0--RC15-blue.svg)](https://plugins.gradle.org/plugin/io.gitlab.arturbosch.detekt)
+[![gradle plugin](https://img.shields.io/badge/Gradle-1.0.0--RC18-blue.svg)](https://plugins.gradle.org/plugin/io.gitlab.arturbosch.detekt)
```
Comment (schalkms) [inline README.md]: RC16

File: detekt-gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -175,3 +190,36 @@ publishing {
...
+
+            gpg(delegateClosureOf<BintrayExtension.GpgConfig> {
+                sign = true
+            })
+
+            mavenCentralSync(delegateClosureOf<BintrayExtension.MavenCentralSyncConfig> {
+                sync = true
+                user = mavenCentralUser
+                password = mavenCentralPassword
+                close = "1"
```
Comment (schalkms) [inline detekt-gradle-plugin/build.gradle.kts]: What's the purpose of this parameter?

File: docs/pages/changelog.md [hunk 3]
```diff
@@ -11,6 +11,49 @@ toc: true
 
 -->
 
+#### RC16
+
+##### Migration
+
+- Notice that we removed the deprecated `warningThreshold` and `failThreshold`. Use `maxIssues` from now one instead.
```
Comment (schalkms) [inline docs/pages/changelog.md]: From now on

File: detekt-gradle-plugin/build.gradle.kts (hunk 2 above)
Comment (arturbosch) [reply detekt-gradle-plugin/build.gradle.kts]: I don't remember anymore :/. I copied it from the main build script.
We could invest in looking at https://github.com/TanVD/kosogor/tree/master/kosogor
However the publish task does not support javadocJar which means we can't sync to maven central anymore ...

**Comment by schalkms** [comment]
I think codefactor warnings should be no reason to fail a pr.

**Pull request merged**